superset/assets/src/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -262,7 +267,14 @@ class SqlEditorLeftBar extends React.PureComponent {
     );
   }
 }
+
+function mapStateToProps({ sqlLab }) {
+  return {
+    offline: sqlLab.offline,
+  };
+}
+
 SqlEditorLeftBar.propTypes = propTypes;
 SqlEditorLeftBar.defaultProps = defaultProps;
 
-export default SqlEditorLeftBar;
+export default connect(mapStateToProps)(SqlEditorLeftBar);
```

Comment:
is adding redux connect for this component necessary? where do you need to check state?

Reply:
it's necessary because I ensure it is online before the ajax calls